graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/SchemaGenerator.kt

```diff
@@ -117,10 +119,15 @@ open class SchemaGenerator(internal val config: SchemaGeneratorConfig) : Closeab
         while (this.additionalTypes.isNotEmpty()) {
             val currentlyProcessedTypes = LinkedHashSet(this.additionalTypes)
             this.additionalTypes.clear()
-            graphqlTypes.addAll(currentlyProcessedTypes.map { generateGraphQLType(this, it.kType, it.inputType) })
+            graphqlTypes.addAll(
+                currentlyProcessedTypes.map {
```

Comment:
i thought our caching would catch duplicates? e.g. it is possible same type could be attempted to be added multiple times

Reply:
The cache does return the value but problem is that now the `additionalTypes` would have two of the type so if you try to add that to the schema builder and then call `.build()` it throws an error

ie

```kotlin
@MyAnnotation
interface Foo

generator. addAdditionalTypesWithAnnotation(MyAnnotation::class, inputType = false)
// 1 additional type 

generator. addAdditionalTypesWithAnnotation(MyAnnotation::class, inputType = true)
// 2 additional types
```

But since it is an interface is is not added as input type so there is a duplicate

Reply:
After further discussion, the issue is with addAdditionalTypesWithAnnotation so I needed to add checks there. This is much better